### PR TITLE
Add "Attach comment to issue" modal

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
1. Create button to open "Attach comment" modal in context menu.
2. Create modal to attach comment.
3. Create API to create issue comment & search issue.
4. Create notification post when comment is attached.

### CODE REVIEW

**Review by Kshitij-Katiyar — CHANGES_REQUESTED** [review]

File: server/gitlab/api.go [hunk 1]
```diff
@@ -658,6 +660,55 @@ func (g *gitlab) CreateIssue(ctx context.Context, user *UserInfo, issue *IssueRe
...
+		issue.IID,
+		&internGitlab.CreateIssueNoteOptions{
+			Body: &issue.Comment,
+		},
+		internGitlab.WithContext(ctx),
+	)
+	if respErr := checkResponse(resp); respErr != nil {
+		return nil, respErr
+	}
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply server/gitlab/api.go:686]: We are checking for the status code of the response in respErr (ie if its 404 or 403).

File: webapp/src/components/attach_comment_to_issue_menu/attach_comment_to_issue.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+
+import GitLabIcon from 'src/images/icons/gitlab';
+
+export default class AttachCommentToIssuePostMenuAction extends PureComponent {
+    static propTypes = {
+        open: PropTypes.func.isRequired,
+        postId: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply webapp/src/components/attach_comment_to_issue_menu/attach_comment_to_issue.jsx:12]: its coming in the props with this name.

**Review by Kshitij-Katiyar — CHANGES_REQUESTED** [review]

File: server/plugin.go [hunk 3]
```diff
@@ -595,3 +595,24 @@ func (p *Plugin) HasGroupHook(ctx context.Context, user *gitlab.UserInfo, namesp
...
 	return found, err
 }
+
+// getUsername returns the GitLab username for a given Mattermost user,
+// if the user is connected to GitLab via this plugin.
+// Otherwise it return the Mattermost username. It will be escaped via backticks.
+func (p *Plugin) getUsername(userID string) (string, error) {
+	info, err := p.getGitlabUserInfoByMattermostID(userID)
+	if err != nil {
+		if err.ID != APIErrorIDNotConnected {
```
[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply server/plugin.go:605]: its a constant already declared in plugin package.....returned by "getGitlabUserInfoByMattermostID" if user is not connected.

**Comment by raghavaggarwal2308** [comment]
Fixes of this PR are in the following PR-

https://github.com/Brightscout/mattermost-plugin-gitlab/pull/10

**Pull request closed without merging**